### PR TITLE
2 hz multilayergraphs OrdMlWD

### LINKED ISSUE #2: Multilayer Graphs
Create:
- [ ] _MultilayerWD.gen.m (Blanca)
- [x] #12 (Hang)
- [x] _MultilayerBD.gen.m (Blanca)
- [x] #19

Create:
- [x] _OrdMlWD.gen.m (Hang)
- [x] _OrdMlWU.gen.m (Hang)
- [ ] _OrdMlBD.gen.m (Blanca)
- [ ] _OrdMlBU.gen.m (jws)

Create
- [x] _OrdMxBUD.gen.m (Hang)
- [x] _OrdMxBUT.gen.m (Blanca)
- [x] #11 (jws)
- [x] _MultilayerBUT.gen.m (Blanca)
- [ ] _OrdMlBUD.gen.m (jws)
- [ ] _OrdMlBUT.gen.m (jws)

### PR DESCRIPTION
**Pull request by hanzh346**
Add OrdMultilayer_WD.gen.m and add the graph to distance measure.

Also add OrdMultilayer_WD.gen.m into temp.m for testing

Unitest and GUI are passed.

### CODE REVIEW

**Review by bzufiriaki — APPROVED** [review]

File: temp.m
Comment (iok-ui) [inline temp.m:1]: the file should not be updated, according to last discussion in nexus (GV, YC, HZ, EG, AC, BZ)

File: braph2genesis/graphs/_OrdMlWD.gen.m [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+B{1,1} = A1;
+B{2,2} = A2;
+B{3,3} = A3;
+B{1,2} = A12;
+B{1,3} = A13;
+B{2,3} = A23;
+B{2,1} = A21;
+B{3,1} = A31;
+B{3,2} = A32;
+A = B
```
Comment (iok-ui) [inline braph2genesis/graphs/_OrdMlWD.gen.m]: leave a space after the comman, e.g. B{3, 2} = A32;
And spread this changes to all the scripts

**Review by iok-ui — CHANGES_REQUESTED** [review]

**Review by JiaweiSun0512 — APPROVED** [review]

File: braph2genesis/graphs/_MultilayerWD.gen.m [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+%%%% ¡calculate!
+if isempty(varargin)
+    layernumber = 1;
+else
+    layernumber = varargin{1};
+end
+value = Graph.NONNEGATIVE * ones(layernumber);
+
+%%% ¡prop!
+A (result, cell) is the cell containing the multilayer weighted adjacency matrices of the multilayer weighted directed graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWD.gen.m]: A (result, cell) is the cell containing the weighted supra-adjacency matrix of the multilayer weighted directed graph.

**Comment by egolol** [comment]
I can't check it, since i get an error when running braph2genesis. See image: 
image

**Comment by hanzh346** [comment]
> I can't check it, since i get an error when running braph2genesis. See image: image

Hello Emiliano,

I just tried with it myself, I didnt get the error.

One think you could do is to exclude "braph2" folder from your path before you compile, that raised errors sometimes.

**Comment by giovannivolpe** [comment]
This is most likely due to the fact that there are still some objects using ConcreteElement in the workspace.
Try to first ensure that the workspace is cleared.

**Review by iok-ui — APPROVED** [review]

File: temp.m
Comment (iok-ui) [inline temp.m:1]: Remove the changes in this temp.m per the discussion before.

**Review by bzufiriaki — APPROVED** [review]

File: braph2genesis/graphs/_MultilayerWD.gen.m [hunk 3]
```diff
@@ -0,0 +1,279 @@
+%% ¡header!
+MultilayerWD < Graph (g, multilayer weighted directed graph) is a multilayer weighted directed graph.
+
+%%% ¡description!
+In a multilayer weighted directed (WD) graph, layers could have different number 
+ of nodes with within-layer weighted directed edges, associated with a real 
+ number between 0 and 1 and indicating the strength of the connection.
+There are connections between layers connecting the corresponding nodes.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWD.gen.m:8]: All node connections are allowed between layers.

File: braph2genesis/graphs/_MultilayerWD.gen.m [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+%%% ¡prop!
+
+NAME (constant, string) is the name of the multilayer weighted directed graph.
+%%%% ¡default!
+'MultilayerWD'
+
+%%% ¡prop!
+DESCRIPTION (constant, string) is the description of the multilayer weighted directed graph.
+%%%% ¡default!
+'In a multilayer weighted directed (BD) graph, layers could have different number of nodes with within-layer weighted directed edges, associated with a real number between 0 and 1 and indicating the strength of the connection. There are connections between layers connecting the corresponding nodes.'
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWD.gen.m:21]: All node connections are allowed between layers.

**Review by annacanal — CHANGES_REQUESTED** [review]

**Comment by hanzh346** [comment]
this pull request is now merged to another one "2 hz multilayerWU OrdMLWU OrdMxBUD OrdMlWD" https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/pull/9

**Pull request closed without merging**